src/ipc/ipc4/helper.c

```diff
@@ -479,7 +479,7 @@ static struct comp_dev *ipc4_create_dai(struct pipeline *pipe, uint32_t id, uint
 static void construct_config(struct ipc4_copier_module_cfg *copier_cfg, uint32_t fifo_size,
 			     struct sof_ipc_stream_params *params, uint32_t scs)
 {
-	uint32_t frame_size;
```

Comment:
@RanderWang A better fix would be to change this function to int and return -EINVAL if none of the fifo_size modulo checks match. But I don't know if that's a possible scenario, so here's a minimum change proposal.

Reply:
yes, I agree with you. recommend fix
````C
    if (fifo_size % 48 == 0) {
                copier_cfg->base.audio_fmt.sampling_frequency = IPC4_FS_48000HZ;
                frame_size = fifo_size / 48;
        } else if (fifo_size % 44 == 0) {
                copier_cfg->base.audio_fmt.sampling_frequency = IPC4_FS_44100HZ;
                frame_size = fifo_size / 44;
        } else if (fifo_size % 32 == 0) {
                copier_cfg->base.audio_fmt.sampling_frequency = IPC4_FS_32000HZ;
                frame_size = fifo_size / 32;
        } else if (fifo_size % 16 == 0) {
                copier_cfg->base.audio_fmt.sampling_frequency = IPC4_FS_16000HZ;
                frame_size = fifo_size / 16;
 +       } else {
 +              tr_err(&comp_tr, "unsupport fifo_size %d" fifo_size);
 +              return -EINVAL;
 +       }
````

Reply:
Thanks, I did it that way